docs/UsingCppInterOp.rst

```diff
@@ -65,35 +65,258 @@ robust (simple function calls, no inheritance, etc.). The goal is to make it as
+only what is necessary. It requires a compiler (Cling or Clang-REPL) that can
+be available during program runtime.
+
+**Reducing Dependencies**
+
+Recent work done on cppyy has been focused on reducing dependencies on
+domain-specific infrastructure (e.g., the ROOT framework). Using an independent
+library such as CppInterOp helps accomplish that, while also reducing the code
+bloat in cppyy.
+
+The CppInterOp library can be configured to use the newly developed Clang-Repl
+backend available in LLVM upstream (or to use the Cling legacy backend, for
+compatibility with High Energy Physics applications).
+
+Only a small set of APIs are needed to connect to the interpreter (Clang-Repl/
+Cling), since other APIs are already available in the standard compiler. This
+is one of the reasons that led to the creation of CppInterOp (a library of
+helper functions), that can help extract out things that are unnecessary for
+for core cppyy functionality.
+
+The cppyy API surface is now incomparably smaller and simpler than what it used
+to be.
+
+**LLVM Integration**
 
 Once CppInterOp is integrated with LLVM's Clang-REPL component (that can then
-be used as a runtime compiler for CPPYY), it will further enhance CPPYY's
+be used as a runtime compiler for cppyy), it will further enhance cppyy's
 performance in the following ways:
 
 
-**Simpler codebase:** The removal of string parsing logic will lead to a simpler
-code base.
+- *Simpler codebase:* The removal of string parsing logic will lead to a
+  simpler code base.
+
+- *Built into the LLVM toolchain:* The CppInterOp interfaces will be part of
+  the LLVM toolchain (as part of Clang-REPL).
+
+- *Better C++ Support:* Finer-grained control over template instantiation is
+  available through CppInterOp.
+
+- *Fewer Lines of Code:* A lot of dependencies and workarounds will be
+  removed, reducing the lines of code required to execute cppyy.
+
+- *Well tested interoperability Layer:* The CppInterOp interfaces have full
+  unit test coverage.
+
+**Making C++ More Social**
+
+cppyy is the first use case demonstrating how CppInterOp can enable C++ to be
+more easily interoperable with other languages. This helps many data scientists
+that are working with legacy C++ code and would like to use simpler, more
+interactive languages.
+
+The goal of these enhancements is to eventually land these interoperability
+tools (including CppInterOp) to greater communities like LLVM and Clang, to
+enable C++ to interact with other languages besides Python.
+
+Example: Template Instantiation
+-------------------------------
+
+The developmental cppyy version can run basic examples such as the one
+here. Features such as standalone functions and basic classes are also
+supported.
+
+C++ code (Tmpl.h)
+
+::
+
+   template <typename T>
+   struct Tmpl {
+     T m_num;
+     T add (T n) {
+       return m_num + n;
+   }
+   };
+
+Python Interpreter
+
+::
+
+   >>> import cppyy
+   >>> import cppyy.gbl as Cpp
```

Comment:
Do we need this line? Did you test that example?

Reply:
Just checked, the example works without this line, removed

Background: I copied it as-is from the following ACAT22 Presentation (slide 9), which may have been built on top of the example in slide 3 (which does require this line of code) 

https://indico.cern.ch/event/1106990/contributions/4991292/attachments/2534654/4363310/ACAT22-BKundu.pdf